### DISCUSSION TITLE
Borg remote backup does not initialize public key, if set up while containers are running

### DISCUSSION THREAD

**Discussion by frederikb96** [nextcloud/all-in-one#5665] [Questions] [answered]
First of all, thank you very much for the nice new remote feature. I discovered this bug:

### Steps to reproduce
0. Containers are online and running

1. No backup configured yet, I will configure it now 
![Image](https://github.com/user-attachments/assets/d49c6886-4ab3-4583-b819-0fd12eb42a70)

2. Insert the url from borgbase and click submit
![Image](https://github.com/user-attachments/assets/9967f769-aac7-4230-9ace-6478b7856493)

3.No public ssh key is shown: 
![Image](https://github.com/user-attachments/assets/8a4f36b9-6fca-4d34-9b7f-21ed015abbfa)

4. I also checked the log, this is still my log from my previous last run with my local backup, so old log, nothing new here

5. I also checked the volume, and no id_borg there yet

### How it worked anyways

I clicked on create backup again, which stops the containers, and then it initializes the repository and I get the pub key. I assume, the problem is that it does not initialize the backup when the containers are running when backup repo is set up :)

**Comment by szaimen** [comment]
Cc @timdiels

**Comment by phillipplum** [comment]
Try looking in the backup logs, the first thing it says is: "Your public ssh key for borgbackup is: xxx"

**Comment by frederikb96** [comment]
Thanks for trying :) But I wrote that I did  that and log showed that actually nothing happened, no Borg container started at all, the GUI button click simply didn't trigger correctly if containers are online.

I fixed this easily by just stopping the containers and then registering the backups, so this is not a problem that stops me from using the new feature.

I just opened the issue to help improving edge cases in the future and to make it more usable for others 🙂

**Comment by szaimen** [comment]
Hey, thanks for this but I think this only happens if you formerly used local backup and change to remote. On a new instance this is not happening in my testing...

**Reply by frederikb96** [reply]
OK perfect thanks for testing 👍🙂

**Comment by szaimen** [comment, marked as answer]
A workaround is to remove the borgbackup container and reload the aio interfacd if you run into this state - moving to discussion for now.
